src/quo/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,215 @@
+(ns quo.components.wallet.transaction-progress.view
+  (:require [quo.components.buttons.button.view :as button]
+            [quo.components.icon :as icons]
+            [quo.components.markdown.text :as text]
+            [quo.components.tags.context-tag.view :as context-tag]
+            [quo.components.wallet.confirmation-progress.view :as confirmation-progress]
+            [quo.components.wallet.transaction-progress.style :as style]
+            [quo.foundations.colors :as colors]
+            [quo.theme :as quo.theme]
+            [react-native.core :as rn]
+            [utils.i18n :as i18n]))
+
+(defn- icon-internal
+  ([icon]
+   (icon-internal icon nil 20))
+  ([icon color]
+   (icon-internal icon color 20))
+  ([icon color size]
+   [rn/view {:style style/icon}
+    [icons/icon
+     icon
+     {:color    color
+      :size     size
+      :no-color (when (not color) true)}]]))
+
+(defn- text-internal
+  [title
+   {:keys [weight size accessibility-label color]
+    :or   {weight :semi-bold
+           size   :paragraph-1}}]
+  [text/text
+   {:accessibility-label accessibility-label
+    :ellipsize-mode      :tail
+    :number-of-lines     1
+    :weight              weight
+    :size                size
+    :style               {:color color}}
+   title])
+
+(defn- network-type-text
+  [state]
+  (cond
+    (or (= state :sending) (= state :pending))      (i18n/label :t/pending-on)
+    (or (= state :confirmed) (= state :finalising)) (i18n/label :t/confirmed-on)
+    (= state :finalized)                            (i18n/label :t/finalized-on)
+    (= state :error)                                (i18n/label :t/failed-on)))
+
+(defn- calculate-counter
+  [counter]
+  (if (< counter 4) counter "4"))
+
+(defn- text-steps
+  [network state epoch-number counter]
+  (let [steps (case network
+                :mainnet
+                {:pending    "0/4"
+                 :sending    (str (calculate-counter counter) "/4")
+                 :confirmed  (str (calculate-counter counter) "/4")
+                 :finalising (str (calculate-counter counter) "/4")
+                 :finalized  (i18n/label :t/epoch-number {:number epoch-number})
+                 :error      "0/4"}
+                (or :optimism :arbitrum :optimism-arbitrum)
```

Comment:
I think we don't have `:optimism-arbitrum` anymore. Am I right?

Reply:
We can remove it and show the networks that are passed in `:networks`. That's a better idea 👍

Reply:
Done